File: distribution/python/dqops/client/models/rule_list_model.py
from typing import Any, Dict, List, Type, TypeVar, Union

from attrs import define as _attrs_define
from attrs import field as _attrs_field

from ..types import UNSET, Unset

T = TypeVar("T", bound="RuleListModel")


@_attrs_define
class RuleListModel:
    """Rule list model

    Attributes:
        rule_name (Union[Unset, str]): Rule name without the folder.
        full_rule_name (Union[Unset, str]): Full rule name, including the folder within the "rules" rule folder.
        custom (Union[Unset, bool]): This rule has is a custom rule or was customized by the user. This is a read-only
            value.
        built_in (Union[Unset, bool]): This rule is provided with DQOps as a built-in rule. This is a read-only value.
        can_edit (Union[Unset, bool]): Boolean flag that decides if the current user can update or delete this object.
        yaml_parsing_error (Union[Unset, str]): Optional parsing error that was captured when parsing the YAML file.
            This field is null when the YAML file is valid. If an error was captured, this field returns the file parsing
            error message and the file location.
    """

    rule_name: Union[Unset, str] = UNSET
    full_rule_name: Union[Unset, str] = UNSET
    custom: Union[Unset, bool] = UNSET
    built_in: Union[Unset, bool] = UNSET
    can_edit: Union[Unset, bool] = UNSET
    yaml_parsing_error: Union[Unset, str] = UNSET
    additional_properties: Dict[str, Any] = _attrs_field(init=False, factory=dict)

    def to_dict(self) -> Dict[str, Any]:
        rule_name = self.rule_name
        full_rule_name = self.full_rule_name
        custom = self.custom
        built_in = self.built_in
        can_edit = self.can_edit
        yaml_parsing_error = self.yaml_parsing_error

        field_dict: Dict[str, Any] = {}
        field_dict.update(self.additional_properties)
        field_dict.update({})
        if rule_name is not UNSET:
            field_dict["rule_name"] = rule_name
        if full_rule_name is not UNSET:
            field_dict["full_rule_name"] = full_rule_name
        if custom is not UNSET:
            field_dict["custom"] = custom
        if built_in is not UNSET:
            field_dict["built_in"] = built_in
        if can_edit is not UNSET:
            field_dict["can_edit"] = can_edit
        if yaml_parsing_error is not UNSET:
            field_dict["yaml_parsing_error"] = yaml_parsing_error

        return field_dict

    @classmethod
    def from_dict(cls: Type[T], src_dict: Dict[str, Any]) -> T:
        d = src_dict.copy()
        rule_name = d.pop("rule_name", UNSET)

        full_rule_name = d.pop("full_rule_name", UNSET)

        custom = d.pop("custom", UNSET)

        built_in = d.pop("built_in", UNSET)

        can_edit = d.pop("can_edit", UNSET)

        yaml_parsing_error = d.pop("yaml_parsing_error", UNSET)

        rule_list_model = cls(
            rule_name=rule_name,
            full_rule_name=full_rule_name,
            custom=custom,
            built_in=built_in,
            can_edit=can_edit,
            yaml_parsing_error=yaml_parsing_error,
        )

        rule_list_model.additional_properties = d
        return rule_list_model

    @property
    def additional_keys(self) -> List[str]:
        return list(self.additional_properties.keys())

    def __getitem__(self, key: str) -> Any:
        return self.additional_properties[key]

    def __setitem__(self, key: str, value: Any) -> None:
        self.additional_properties[key] = value

    def __delitem__(self, key: str) -> None:
        del self.additional_properties[key]

    def __contains__(self, key: str) -> bool:
        return key in self.additional_properties

File: dqops/src/main/java/com/dqops/rest/models/metadata/RuleListModel.java
/*
 * Copyright © 2021-Present DQOps, Documati sp. z o.o. (support@dqops.com)
 *
 * This file is licensed under the Business Source License 1.1,
 * which can be found in the root directory of this repository.
 *
 * Change Date: This file will be licensed under the Apache License, Version 2.0,
 * four (4) years from its last modification date.
 */
package com.dqops.rest.models.metadata;

import com.dqops.utils.docs.generators.SampleStringsRegistry;
import com.dqops.utils.docs.generators.SampleValueFactory;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonPropertyDescription;
import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import io.swagger.annotations.ApiModel;
import lombok.Data;

/**
 * Rule list model that is returned by the REST API.
 */
@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
@ApiModel(value = "RuleListModel", description = "Rule list model")
public class RuleListModel {
    /**
     * Rule name.
     */
    @JsonPropertyDescription("Rule name without the folder.")
    private String ruleName;

    /**
     * Full rule name, including the folder within the "rules" rule folder.
     */
    @JsonPropertyDescription("Full rule name, including the folder within the \"rules\" rule folder.")
    private String fullRuleName;

    /**
     * This rule has is a custom rule or was customized by the user. This is a read-only value.
     */
    @JsonPropertyDescription("This rule has is a custom rule or was customized by the user. This is a read-only value.")
    private boolean custom;

    /**
     * This rule is provided with DQOps as a built-in rule. This is a read-only value.
     */
    @JsonPropertyDescription("This rule is provided with DQOps as a built-in rule. This is a read-only value.")
    private boolean builtIn;

    /**
     * Boolean flag that decides if the current user can update or delete this object.
     */
    @JsonPropertyDescription("Boolean flag that decides if the current user can update or delete this object.")
    private boolean canEdit;

    /**
     * Optional parsing error that was captured when parsing the YAML file.
     * This field is null when the YAML file is valid. If an error was captured, this field returns the file parsing error message and the file location.
     */
    @JsonPropertyDescription("Optional parsing error that was captured when parsing the YAML file. " +
            "This field is null when the YAML file is valid. If an error was captured, this field returns the file parsing error message and the file location.")
    private String yamlParsingError;

    public RuleListModel() {
    }

    public static class RuleListModelSampleFactory implements SampleValueFactory<RuleListModel> {
        @Override
        public RuleListModel createSample() {
            return new RuleListModel() {{
                setRuleName(SampleStringsRegistry.getRuleName());
                setFullRuleName(SampleStringsRegistry.getFullRuleName());
                setCustom(false);
                setBuiltIn(true);
                setCanEdit(true);
            }};
        }
    }
}
